### ISSUE TITLE
Empty string "" denormalize function fails to denormalize properly

### ISSUE THREAD

**Issue by anton6** [paularmstrong/normalizr#366]
# Version: 3.3.0

# Problem

When key is empty string "" **denormalize(input, schema, entities)** function fails to denromalize the data properly.

This has already been discussed here: **https://github.com/paularmstrong/normalizr/issues/278**

Seems it's generally fails to correctly denormalize for any falsy values, including **null**

**Comment by Eterion** [comment]
Indeed that is the case, here's example:

```js
const { normalize, denormalize, schema } = require("normalizr");

const data = [
    {
        _uniqueId: 0,
        type: "a",
        name: "First",
    },
    {
        _uniqueId: 1,
        type: "b",
        name: "Second",
    },
    {
        _uniqueId: 2,
        type: "c",
        name: "Third",
    },
];

const arraySchema = new schema.Array({
    a: new schema.Entity("a", {}, { idAttribute: "_uniqueId" }),
    b: new schema.Entity("b", {}, { idAttribute: "_uniqueId" }),
    c: new schema.Entity("c", {}, { idAttribute: "_uniqueId" }),
}, value => value.type);

const normalized = normalize(data, arraySchema);
const denormalized = denormalize(normalized.result, arraySchema, normalized.entities);
console.log(denormalized);
```

logs

```js
[
  {
    id: 0
    schema: "a"
  },
  {
    _uniqueId: 1
    name: "Second"
    type: "b"
  },
  {
    _uniqueId: 2
    name: "Third"
    type: "c"
  },
]
```

**Comment by paularmstrong** [comment]
Fixed in #345. Please wait for the next release or pull from the master branch until then.
